client/lib/signup/step-actions/index.js

```diff
@@ -560,6 +560,13 @@ export function createAccount(
 			return;
 		}
 
+		// Handling special case where users log in via social using signup form.
+		let newAccountCreated = true;
+
+		if ( signupType === SIGNUP_TYPE_SOCIAL && response && ! response.created_account ) {
```

Comment:
We can use optional chaining here:
```suggestion
		if ( signupType === SIGNUP_TYPE_SOCIAL && ! response?.created_account ) {
```